audit_log/managers.py

```diff
@@ -0,0 +1,67 @@
+from typing import Optional, TYPE_CHECKING, Union
+
+from django.db import models
+from django.http import HttpRequest
+
+from audit_log.enums import Operation, Status
+from audit_log.services import audit_log_service
+from audit_log.types import AuditCommitMessage
+from audit_log.utils import create_commit_message, get_remote_address
+
+if TYPE_CHECKING:
+    from django.contrib.auth.models import AbstractUser
+
+
+class AuditLogQuerySet(models.QuerySet):
+    def with_audit_log(
+        self,
+        user: "AbstractUser",
+        operation: Operation,
+        status: Union[Status, str] = Status.SUCCESS.value,
+        ip_address="",
+        path: Optional[str] = None,
+    ):
+        if not user:
+            raise ValueError("User cannot be set to None.")
+
+        if not operation:
+            raise ValueError("Operation cannot be None or an empty string.")
+
+        if not status:
+            raise ValueError("Status cannot be None or an empty string.")
+
+        # Use the model name as a path if it is not given
+        if not path:
+            path = self.model.__name__
+
+        # Get the affected object ids as a list of strings
+        object_ids = [str(pk_tuple[0]) for pk_tuple in self.values_list("pk")]
```

Comment:
`flat` parameter makes this slightly simpler IMHO:
```suggestion
        object_ids = [str(pk) for pk in self.values_list("pk", flat=True)]
```

Reply:
changed